server/src/main/java/io/druid/segment/realtime/RealtimeManager.java

```diff
@@ -322,11 +321,11 @@ private void runFirehoseV2(FirehoseV2 firehose)
           inputRow = firehose.currRow();
           if (inputRow != null) {
             try {
-              numRows = plumber.add(inputRow);
+              numRows = plumber.add(inputRow, committerSupplier);
             }
             catch (IndexSizeExceededException e) {
-              log.debug(e, "Index limit exceeded: %s", e.getMessage());
-              nextFlush = doIncrementalPersist(firehose.makeCommitter(), intermediatePersistPeriod);
+              log.info("Index limit exceeded: %s", e.getMessage());
+              metrics.incrementThrownAway();
```

Comment:
ok, I think I can just remove this entire block; since the plumber is in control of persisting after the index is full, IndexSizeExceededException should not happen. That would then fall through to the 'unknown exception' handler.